bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateService.java

```diff
@@ -67,6 +66,7 @@
  * @author Thomas Höfer - Initial contribution
  * @author Dimitar Ivanov - update and cancel operations are run with different safe caller identifiers in order to
  *         execute asynchronously
+ * @author Dimitar Ivanov - Firmware update is done for thing
```

Comment:
same as with the `FirmwareRegistry`: Could you please turn this class into an interface and move the implementation to `org.eclipse.smarthome.core.thing.internal.firmware.FirmwareUpdateServiceImpl`?